### PR TITLE
Ignore Redis firewall rules for caches using private endpoints

### LINKED ISSUE #2171: Ignore Redis firewall rules for caches using private endpoints
**Description of the issue**

The following rules are not applicable when Redis is configured to allow private connectivity by setting `properties.publicNetworkAccess` to `Disabled`.

- Azure.Redis.FirewallRuleCount
- Azure.Redis.FirewallRuleIPRange

We should update docs for these rules and add a precondition.

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **1.26.1**

**Additional context**

https://learn.microsoft.com/azure/azure-cache-for-redis/cache-network-isolation#limitations-of-firewall-rules

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2171 

Ignore Redis firewall rules for caches using private endpoints.

Updated `Azure.Redis.FirewallRuleCount` and `Azure.Redis.FirewallRuleIPRange`.

Added pre-condition and updated rule docs.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [ ] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
Hi @BernieWhite you let me know.

File: src/PSRule.Rules.Azure/rules/Azure.Redis.Rule.ps1 [hunk 1]
```diff
@@ -181,13 +181,28 @@ function global:IsEnterpriseCache {
...
+    param (
+        
+    )
+    process {
+        if ($PSRule.TargetType -eq 'Microsoft.Cache/redis' -and $TargetObject.properties.publicNetworkAccess -ne 'Disabled') {
+            $True
+        }
+        else {
+            $False
+        }
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.Redis.Rule.ps1:205]: Can be simplified to `return $PSRule.TargetType -eq 'Microsoft.Cache/redis' -and $TargetObject.properties.publicNetworkAccess -ne 'Disabled';` however we also need to check the case when the `Microsoft.Cache/redis/firewallRules` type is used.

If `Microsoft.Cache/redis/firewallRules` is used, we should just check it as normal.

```suggestion
        return $PSRule.TargetType -eq 'Microsoft.Cache/redis/firewallRules' -or ($PSRule.TargetType -eq 'Microsoft.Cache/redis' -and $TargetObject.properties.publicNetworkAccess -ne 'Disabled');
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Nice work @BenjaminEngeset. Also thanks for the documentation updates. They look great. See comments below relating to `Microsoft.Cache/redis/firewallRules`.

Let's add an additional test to check for this case.

**Comment by BenjaminEngeset** [comment]
Thanks @BernieWhite for identifying the bug I was close to implement 😿!

I've added the improved logic in the helper function and added a test case where the public network access is disabled.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. The updates look good. All good to merge.

**Pull request merged**